### ISSUE TITLE
how to calculate loss?

### ISSUE THREAD

**Issue by eeric** [HikariTJU/LD#5]
hi,  was it implemented to losses = self(**data)?
on 234th line in base.py in mmdet/models/detectors

**Comment by HikariTJU** [comment]
mmdet/models/dense_heads/ld.py

**Comment by eeric** [comment]
thanks
no ld.py but ld_head.py.

besides, it couldn't debugged to losses = self(**data) one step, and then didn't know how to calculate LD loss.

**Comment by HikariTJU** [comment]
function loss() and loss_single()  in ld_head.py

**Comment by eeric** [comment]
ok, the wrapping code can look complicated
one sample resolution: 
about GFL arctecture, self.loss_dfl = build_loss(loss_dfl) in ld_head.py, here  build_loss with self.loss_dfl = DistributionFocalLoss(loss_weight=self.loss_cfg.loss_dfl.loss_weight) difference, 119th in
https://github.com/RangiLyu/nanodet/blob/main/nanodet/model/head/gfl_head.py

**Comment by eeric** [comment]
class LDLoss(nn.Module) on 176th line in gfocal_loss.py, that was it used?

**Comment by eeric** [comment]
it didn't employed to class DistributionFocalLoss on 137th line in gfocal_loss.py, did it?

**Comment by HikariTJU** [comment]
不是很明白你的问题

**Comment by eeric** [comment]
你的代码封装的太好了，单步调试没法定位到具体如何计算LD loss的。
所以，我想用个简单办法，即dfl loss不仅仅是它，又加上了ld loss，这样的话，只用到了在 gfocal_loss.py的176行的LDLoss函数吧。

**Comment by HikariTJU** [comment]
嗯 LDLoss 同时算了loss dfl 和 loss ld

**Comment by eeric** [comment]
ok, thanks,
I'll try. wish your best.

**Comment by eeric** [comment]
reduction='mean', loss_weight=1.0, T=2, alpha=1, beta=1 in LDLoss, did it?
